### ISSUE TITLE
`no-unused-vars` & `@typescript-eslint/no-unused-vars` shouldn't be turned off for `.vue` files

### ISSUE THREAD

**Issue by haoqunjiang** [oxc-project/eslint-plugin-oxlint#316]
As the documentation says, the `no-unused-vars` rule in oxlint ["ignores `.d.ts` files and `.vue` files entirely"](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-unused-vars.html#ignored-files).
Therefore this plugin should not turn off `no-unused-vars` and `@typescript-eslint/no-unused-vars` for `.vue` files. Otherwise, there would be false negatives.

Reproduction:

```sh
git clone https://github.com/haoqunjiang/oxlint-unused-vars.git
cd oxlint-unused-vars
pnpm i
pnpm lint # linting passes without catching the unused var in src/App.vue
sed -i '' '/oxlint/s/^/\/\//' eslint.config.js # comment out eslint-plugin-oxlint
pnpm lint # linting fails as it now catches the unused var
```

**Comment by Sysix** [comment]
The problem is that we cant detect variables from the `<script>` tag which are included in `<template>`.
The same problem has the native eslint/typescript rule.

Your example repository includes `eslint-vue-plugin/flat/essential` which includes [vue/no-unused-vars](https://eslint.vuejs.org/rules/no-unused-vars.html). 
This only checks for the `<template>` part of a vue file.

There is no existing rule which currently supports the combination of `<script>` and `<template>`. 
If you still want to fix this use case, then the `oxlint` project is the issue.

**Comment by haoqunjiang** [comment]
While the name might be confusing, `vue/no-unused-vars` is actually for variables *defined* in `v-for` expressions.

What I'm talking about in this issue is the native eslint/typescript `no-unused-vars`, which *can* detect such problems with the help of [`vue-eslint-parser`](https://github.com/vuejs/vue-eslint-parser/blob/bb5863373cef67a364fa26412dae7a4913ab1589/src/script-setup/scope-analyzer.ts#L157-L216).

So I think `eslint-plugin-oxlint` shouldn't override these rules because oxlint's capability doesn't match the native ones' in this case.

**Comment by Sysix** [comment]
Oh. With the parser it is possible to detect it. Thanks for the info.

You are using `oxlint.configs['flat/recommended']` which is just a list of rules. 
We can support this use case only with `buildFromOxlintConfig`. 
This is the function where we can return multiple config arrays. 

POC:
````json
[{
  "name": "oxlint/from-oxlint-config",
  "rules": {
    ... // no no-unused-vars here
  },
}, {
  "name": "oxlint/unused-for-non-vue",
  "files": ["!*.vue"],
  "rules": {
    "no-unused-vars": "off"
  },
}]

````

Removing the rule completly from this plugin is too much. 
Someone who does not use vue should still be able to disable the rule.

Would the POC solve the problem for you?

**Comment by Sysix** [comment]
> I know it would be a breaking change, but the config would still be easy to use, in flat config, users only need to add a spread operator upfront.

We are only allowed to do this with `flat/*` because we are currently supporting eslint v8. 
Maybe we can use `overrides` for them. I do not know internal part of eslint to say if this is safe.

**Comment by DeronEndless** [comment]
<img width="719" alt="Image" src="https://github.com/user-attachments/assets/820c0117-b1a2-44fa-bdff-09ad5af2fe0c" />

**Comment by DeronEndless** [comment]
<img width="719" alt="Image" src="https://github.com/user-attachments/assets/17f2bcd0-8c32-4f85-b872-caac7c450068" />

I have the same problem, how to solve it

**Comment by DeronEndless** [comment]
No problem，I see "https://github.com/vuejs/eslint-config-typescript" update

**Comment by haoqunjiang** [comment]
> We are only allowed to do this with `flat/*` because we are currently supporting eslint v8. Maybe we can use `overrides` for them. I do not know internal part of eslint to say if this is safe.

I think it's safe. `typescript-eslint` has been using `overrides` for a long time: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/eslint-recommended.ts
